be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -313,6 +316,7 @@ void FragmentExecutor::_convert_data_sink_to_operator(PipelineBuilderContext* co
             ops.emplace_back(source_op);
             ops.emplace_back(sink_op);
             auto pp = std::make_shared<Pipeline>(context->next_pipe_id(), ops);
+            pp->set_root();
```

Comment:
Because we append new pipeline to the tail of `_fragment_ctx->pipelines()`,  the prev tail pipeline created by `PipelineBuilder::build()` which contains `MultiCastLocalExchangeSource` is not root anymore.
Could we set it as not root? or maybe it is ok and doesn't hurt performance if we also set the pipeline which contains `MultiCastLocalExchangeSource` as root.

Reply:
I've tested that it can be set as root. Although maybe it hurts some performance, but there is no correctness problem.